ci/Jenkinsfile_utils.groovy

```diff
@@ -112,7 +112,18 @@ def get_git_commit_hash() {
 }
 
 def publish_test_coverage() {
-    sh "curl -s https://codecov.io/bash | bash"
+    // CodeCovs auto detection has trouble with our CIs PR validation due the merging strategy
+    git_commit_hash = get_git_commit_hash()
+
+    if (env.CHANGE_ID) {
+      // PR execution
+      codecovArgs = "-f '*.xml' -B ${env.CHANGE_TARGET} -P ${env.CHANGE_ID} -C ${git_commit_hash}"
```

Comment:
Why is this filter necessary? I think codecov is able to find the reports on its own and also there might be other coverage reports that are not xml Files - like gcov..

Reply:
It's not necessary. I will remove it. 

The strange thing is that the codecoverage xml report was generated during the test, but codecov script didn't find the xml file. 
https://jenkins.mxnet-ci.amazon-ml.com/blue/organizations/jenkins/mxnet-validation%2Funix-cpu/detail/PR-20138/2/pipeline/283

When we use nose to run these tests, there will have seperate xml report files generated, like nosetests_unittest.xml and nosetests_mkl.xml, but the final submitted report is coverage.xml
You can find in
https://jenkins.mxnet-ci.amazon-ml.com/blue/organizations/jenkins/mxnet-validation%2Funix-cpu/detail/PR-19876/1/pipeline/297